hearpreprocess/sampler.py

```diff
@@ -80,10 +83,39 @@ def requires(self):
         return pipeline.get_download_and_extract_tasks(self.task_config)
 
     @staticmethod
-    def safecopy(dst, src):
+    def safecopy(src, dst):
         # Make sure the parent exists
         dst.parent.mkdir(parents=True, exist_ok=True)
-        shutil.copy(src, dst)
+        shutil.copy2(src, dst)
+
+    @staticmethod
+    def trimcopy_audio(src, dst, small_duration):
+        """
+        Trims and saves the audio file to minimise the size of the generated
+        small dataset
+        """
+        # Make sure the parent exists
+        dst.parent.mkdir(parents=True, exist_ok=True)
+        # Trim the audio if it is greater than the small_duration
+        audio_stats = audio_util.get_audio_stats(str(src))
+        if audio_stats is not None and audio_stats["duration"] > small_duration:
```

Comment:
a) don't we have an audioutil.py method to do this instead?
b) shouldn't we convert to WAV first? to keep durations accurate?

Reply:
a)the method in `audioutil.py`, is for trimming and padding, but we don't want to pad in this case, so made this call here.
b) Yes, will add that.

Reply:
@khumairraj still, why not put the trim function in audioutil?

Reply:
used the same functions. I think they will work in this case